01_Introduction/Hello/HelloWinForms/Form1.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Windows.Forms;
+
+namespace HelloWinForms
+{
+    public partial class Form1 : Form
+    {
+        public Form1()
+        {
+            InitializeComponent();
+        }
+
+        private void button1_Click(object sender, EventArgs e)
+        {
+            MessageBox.Show($"Hello, {textBox1.Text}!", "Hello", MessageBoxButtons.OK);
+        }
+
+        private void textBox1_TextChanged(object sender, EventArgs e)
+        {
+            if (string.IsNullOrEmpty(textBox1.Text))
+                button1.Enabled = false;
```

Comment:
It is better to use brackets `{ }`, especially if you are using them in other places, like `else` block below.

Reply:
Please respond to the comments when you fix them. Thanks.

Reply:
OK, I have fixed that.